pyannote/audio/pipelines/speaker_diarization.py

```diff
@@ -338,6 +338,10 @@ def iter_waveform_and_mask():
         for i, batch in enumerate(batches, 1):
             waveforms, masks = zip(*filter(lambda b: b[0] is not None, batch))
 
+            if torch.cuda.is_available():
```

Comment:
I think we should move to GPU only if `.device == 'cuda'`. This can lead to some troubles, for example if user have gpu available, but he wants to run it on cpu. Also it don't match with `.to` behaviour